modules/setting/mailer.go

```diff
@@ -70,9 +70,12 @@ func newMailService() {
 	if sec.HasKey("HOST") && !sec.HasKey("SMTP_ADDR") {
 		givenHost := sec.Key("HOST").String()
 		addr, port, err := net.SplitHostPort(givenHost)
-		if err != nil {
+		if err != nil && err.Error() != "missing port in address" {
 			log.Fatal("Invalid mailer.HOST (%s): %v", givenHost, err)
 		}
+		if addr == "" {
+			addr = "127.0.0.1"
+		}
```

Comment:
If an error is returned then addr and port will be empty.

Therefore when there is an error we should re-set addr to `givenHost`.


```suggestion
		if err != nil && err.Error() == "missing port in address" {
			addr = givenHost   
		} else if err != nil {
			log.Fatal("Invalid mailer.HOST (%s): %v", givenHost, err)
		}
		if addr == "" {
			addr = "127.0.0.1"
		}		
```

I don't think it's necessarily correct to re-set an empty SMTP_ADDR to 127.0.0.1 but I guess it's not necessarily incorrect